### PR TITLE
public @Sendable async throws

### PR DESCRIPTION
**Pull request by Obbut**

### CODE REVIEW

File: Tests/DependenciesMacrosTests/DependenciesMacrosTests.swift [hunk 1]
```diff
@@ -48,20 +48,335 @@ final class DependenciesMacrosTests: XCTestCase {
...
+            final class MyDependency {
+                func foo(bar: String, baz: Int) -> String {
+                    return bar
+                }
+            }
+            protocol MyDependencyProtocol : AnyObject {
+                func foo(bar: String, baz: Int) -> String
+            }
+            class MyDependencyMock: MyDependencyProtocol {
+                var _foo: (String, Int) -> String = unimplemented()
```
Comment (Joannis) [inline Tests/DependenciesMacrosTests/DependenciesMacrosTests.swift:193]: Just wondering, but what's the use of this default value i the `init` already overrides it with a default?

**Review by Joannis — APPROVED** [review]
LGTM

File: Tests/DependenciesMacrosTests/DependenciesMacrosTests.swift (hunk 1 above)
Comment (Obbut) [reply Tests/DependenciesMacrosTests/DependenciesMacrosTests.swift:193]: Absolutely nothing at all 😉

Comment (Obbut) [reply Tests/DependenciesMacrosTests/DependenciesMacrosTests.swift:193]: Unresolved because I'll remove this 🙃

**Pull request closed without merging**